src/GNDStk/Node/test/all.test.cpp

```diff
@@ -1,26 +1,96 @@
 
 #include "catch.hpp"
 #include "GNDStk.hpp"
+#include <cstring>
 
 using namespace njoy::GNDStk;
 using namespace mixed::child;
+using namespace mixed::meta;
 
 SCENARIO("Testing GNDStk Node all()") {
 
    GIVEN("A tree read from a particular example GNDS file") {
       // tree
       const Tree<> tree("n-008_O_016.xml");
 
-      // node: chemicalElements
-      const auto &elems = tree(reactionSuite,PoPs,chemicalElements);
+      // chemicalElements / chemicalElement
+      WHEN("We extract all <chemicalElement> nodes within <chemicalElements>") {
+         const auto &CEs = tree(reactionSuite,PoPs,chemicalElements);
+         {
+            // without "found"
+            auto ce = CEs.all("chemicalElement");
+            THEN("We find eight of them") {
+               CHECK(ce.size() == 8);
+            }
+         } {
+            // with "found"
+            bool found = false;
+            auto ce = CEs.all("chemicalElement",found);
+            THEN("We find eight of them") {
+               CHECK(found);
+               CHECK(ce.size() == 8);
+            }
+         }
+      }
+
+      // reactions / reaction
+      WHEN("We extract selected <reactions> nodes within <reaction>") {
+         const auto &Rs = tree(reactionSuite,reactions);
+
+         // filter that requires: node has metadata label="n + *"
+         auto nplus = [](const node &n)
+            { return 0 == strncmp(n(label).c_str(), "n + ", 4); };
+
+         // filter that requires: node has metadata ENDF_MT="600"
+         auto endf_mt_600 = [](const node &n)
+            { return n(ENDF_MT) == 600; };
+
+         // filter that requires: both of the above
+         auto filter_both = [nplus,endf_mt_600](const node &n)
+            { return nplus(n) && endf_mt_600(n); };
+
+         THEN("There are 15 <reaction> nodes with label=\"n + *\"") {
```

Comment:
I like this idea of a filter. I think it will be very helpful.

Reply:
Definitely. I anticipate that a lot of other capabilities will be built upon this. There's a bit more I have in mind to do regarding filters, but there are probably other priorities at the moment.